snaps/learn/tutorials/gas-estimation.md

```diff
@@ -28,103 +28,146 @@ starter kit by running:
+    ```
+
+    Download
+    [this `gas.svg` icon file](https://raw.githubusercontent.com/Montoya/gas-fee-snap/main/packages/snap/images/gas.svg)
+    into that `ìmages` folder.  
+
+    <details>
+      <summary>Icon attribution</summary>
+      <div>
+    This is a free icon, "Gas" by Mello from the [Noun Project](https://thenounproject.com/browse/icons/term/gas/).
+     </div>
+    </details>
+
+
+    Your file structure should look like this:
+
+    ```text
+    gas-estimation-snap/
+    ├─ packages/
+    │  ├─ site/
+    |  |  |- src/
+    |  |  |  |- App.tsx
+    |  |  ├─ package.json
+    |  |  |- ...(react app content)
+    |  |
+    │  ├─ snap/
+    |  |  ├─ images/
+    |  |  |  |- gas.svg
+    |  |  ├─ src/
+    |  |  |  |- index.test.ts
+    |  |  |  |- index.ts
+    |  |  ├─ snap.manifest.json
+    |  |  ├─ package.json
+    |  |  |- ... (Snap content)
+    ├─ package.json
+    ├─ ... (other stuff)
+    ```
+
+    Open `packages/snap/snap.manifest.json` in a text editor. This file contains the main configuration 
+    details for your Snap. Edit the `npm` object, within the `location` object, and add `iconPath` with 
+    the value `"images/gas.svg"` to point to your new icon:
+
+    ```json title="snap.manifest.json"
+    "location": {
+      "npm": {
+        "filePath": "dist/bundle.js",
+        "iconPath": "images/gas.svg",
+        "packageName": "snap",
+        "registry": "https://registry.npmjs.org/"
+      }
+    }
+    ```
 
-### 2. Add a custom icon
+    Open `packages/snap/package.json` in a text editor. Edit the `files` array and reference the 
+    `images/` folder:
 
-Your Snap must [display an icon](../best-practices/design-guidelines.md#optimize-your-metadata) in MetaMask.
-To add an icon, create a new folder `images` in the Snap package `packages/snap/`:
+    ```json title="package.json"
+    "files": [
+      "dist/",
+      "images/",
+      "snap.manifest.json"
+    ],
+    ```
 
-```bash
-mkdir packages/snap/images
-```
+    2.2 Update your Snap's wallet prompt
 
-Download
-[this `gas.svg` icon file](https://raw.githubusercontent.com/Montoya/gas-fee-snap/main/packages/snap/images/gas.svg)
-into the `ìmages` folder you just created.  
-This is a free icon, "Gas" by Mello from
-[Noun Project](https://thenounproject.com/browse/icons/term/gas/).
-
-Your file structure should look like this:
-
-```text
-gas-estimation-snap/
-├─ packages/
-│  ├─ site/
-|  |  |- src/
-|  |  |  |- App.tsx
-|  |  ├─ package.json
-|  |  |- ...(react app content)
-|  |
-│  ├─ snap/
-|  |  ├─ images/
-|  |  |  |- gas.svg
-|  |  ├─ src/
-|  |  |  |- index.test.ts
-|  |  |  |- index.ts
-|  |  ├─ snap.manifest.json
-|  |  ├─ package.json
-|  |  |- ... (Snap content)
-├─ package.json
-├─ ... (other stuff)
-```
+    It's important that user's understand what they're agreeing to when responding to wallet prompts. The 
+    prompt uses the `proposedName` of the Snap, currently "Typescript Example" in our boilerplate.
```

Comment:
```suggestion
    It's important that users understand what they're agreeing to when responding to wallet prompts. The 
    prompt uses the `proposedName` of the Snap, currently "Typescript Example" in the boilerplate.
```

I also wonder about the use of "boilerplate," and if "template" would be a more accessible term.